src/sentry/sentry_metrics/consumers/indexer/batch.py

```diff
@@ -146,8 +163,8 @@ def filter_messages(self, keys_to_remove: Sequence[PartitionIdxOffset]) -> None:
         self.skipped_offsets.update(keys_to_remove)
 
     @metrics.wraps("process_messages.extract_strings")
-    def extract_strings(self) -> Mapping[int, Set[str]]:
-        org_strings = defaultdict(set)
+    def extract_strings(self) -> Mapping[str, Mapping[int, Set[str]]]:
```

Comment:
`Mapping[str, Mapping[int, Set[str]]]:`  starts becoming a very opaque type, this makes it really hard to understand how it is supposed to be used (specifically what the key means) Consider adding type aliases. 
Example: you can assign an alias to str there `UseCaseId` if we do not have an enum for it.